### PR TITLE
pin azure to a stable version

### PR DESCRIPTION
**Pull request by danrobinson88**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -16,7 +16,11 @@ requests = "*"
 redis = {version=">=3.3.0", optional=true}
 pika = {version="*", optional=true}
 kafka-python = {version="*", optional=true}
-azure-storage-blob = { version = "^12.0.0", optional = true }
+
+# The version is pinned because 12.7.0 is broken, remove the pinning after it will be fixed.
+# github issue: https://github.com/Azure/azure-sdk-for-python/issues/16193
+azure-storage-blob = { version = "12.6.0", optional = true }
```
Comment (aviramha) [inline pyproject.toml:22]: I think we should be more specific
`"^12.0.0, != 12.7.0"`

Comment (aviramha) [reply pyproject.toml:22]: I meant in the actual specifications, not in the documentation :)

File: pyproject.toml [hunk 2]
```diff
@@ -16,7 +16,11 @@ requests = "*"
 redis = {version=">=3.3.0", optional=true}
 pika = {version="*", optional=true}
 kafka-python = {version="*", optional=true}
-azure-storage-blob = { version = "^12.0.0", optional = true }
+
+# The version is pinned because 12.7.0 breaks, remove the pinning after it will be fixed.
```
Comment (bentheiii) [inline pyproject.toml:20]:
```suggestion
# 12.7.0 is yanked because it is broken for azurite.
```

**Review by bentheiii — CHANGES_REQUESTED** [review]
* document the change in the changelog
* the version also needs to be corrected in _version.py

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,7 @@
 # Yellowbox Changelog
+## 0.4.1
+### Changed
```
Comment (bentheiii) [inline CHANGELOG.md:3]: This is a fix-level change

```suggestion
### Fixed
```

**Review by aviramha — APPROVED** [review]

**Comment by aviramha** [comment]
Thanks!

**Pull request merged**